prow/sidecar/run_test.go

```diff
@@ -528,3 +535,74 @@ func TestLogReaders(t *testing.T) {
 	}
 
 }
+
+func TestSideCarLogsUpload(t *testing.T) {
+	logFile, err := LogSetup()
+	if err != nil {
+		t.Fatalf("Unable to set up log file")
+	}
+	testString := "Testing...Hello world!"
+	logrus.Info(testString)
+	var once sync.Once
+
+	localOutputDir, err := ioutil.TempDir("", "testdir")
+	if err != nil {
+		t.Errorf("Unable to create temp dir: %v", err)
+	}
```

Comment:
Please defer an os.RemoveAll of the temp directory so that it gets cleaned up when the tests are done.